utils/mfasweep.py

```diff
@@ -0,0 +1,156 @@
+from datetime import datetime
+import adal #https://adal-python.readthedocs.io/en/latest/
+import msal
+import pprint
+import utils.utils as utils
+import utils.notify as notify
+
+
+# MFA Sweep Functions
+# MFA Sweep Code to pull token and check endpoints
+def _input_scopes():
+    return _select_options(
+        [
+            "https://graph.microsoft.com/.default",
+            "https://management.azure.com/.default",
+            "User.Read",
+            "User.ReadBasic.All",
+        ],
+        header="Select a scope (multiple scopes can only be input by manually typing them, delimited by space):",
+        accept_nonempty_string=True,
+    ).split()
+
+
+# ​
+def acquire_token_by_username_password(app, username, password):
+    """acquire_token_by_username_password() - See constraints here: https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-authentication-flows#constraints-for-ropc"""
+    pprint.pprint(
+        app.acquire_token_by_username_password(
+            username, password, scopes=_input_scopes(), app_name="My Signins"
+        )
+    )
+
+
+_JWK1 = """{"kty":"RSA", "n":"2tNr73xwcj6lH7bqRZrFzgSLj7OeLfbn8216uOMDHuaZ6TEUBDN8Uz0ve8jAlKsP9CQFCSVoSNovdE-fs7c15MxEGHjDcNKLWonznximj8pDGZQjVdfK-7mG6P6z-lgVcLuYu5JcWU_PeEqIKg5llOaz-qeQ4LEDS4T1D2qWRGpAra4rJX1-kmrWmX_XIamq30C9EIO0gGuT4rc2hJBWQ-4-FnE1NXmy125wfT3NdotAJGq5lMIfhjfglDbJCwhc8Oe17ORjO3FsB5CLuBRpYmP7Nzn66lRY3Fe11Xz8AEBl3anKFSJcTvlMnFtu3EpD-eiaHfTgRBU7CztGQqVbiQ", "e":"AQAB"}"""
```

Comment:
Are these ever used? I don't see them anywhere in the code, curious what they are

Reply:
Yes, when the function gets a valid set of creds that lack MFA they're used when a token is requested in the ADAL lib, I couldn't get it to work without them

Reply:
dafuqqq that's wild hah